wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Add site list to Welcome Screen (3)

Fixes Partially #17368
Depends on: #17388, which depends on #17387
`Ready for merge` after #17388

Adds the sites list to the Welcome Screen of the Jetpack Migration Flow, which is behind a temporary feature flag.

To test:

0. **Prerequisite:** Toggle the following feature flags `on` in `Wordpress/build.gradle` code:
   (it's not possible to do this from the app because we need to be logged in to switch feature flags):
   - Set `JETPACK_SHARED_LOGIN` to `true`
   - Set `JETPACK_PROVIDER_SYNC` to `true`
   - Set `JETPACK_MIGRATION_FLOW` to `true`
   - Press the `Sync Now` button that appears at the top right of the file editing view in Android Studio
1. Build & Install the WordPress app then Login
   _Note: If testing this right after [dependency PR (2)](https://github.com/wordpress-mobile/WordPress-Android/pull/17388) it's enough to uninstall the Jetpack app and skip step 1._
2. Build & Install the Jetpack app **using the same variant**
   (e.g. if WordPress was `wordpressWasabiDebug` , use `jetpackWasabiDebug`)
3. **Expect** to see the "Welcome to Jetpack!" screen
4. **Verify** that the UI matches the design
   ℹ️ Note: The blur effect behind the buttons is [only available on Android 12 or higher](https://github.com/wordpress-mobile/WordPress-Android/pull/17389#pullrequestreview-1167090705).
   _Note: There's been some recent design updates which are not implemented in this PR._
   1. Scroll down
   2. **Expect** the header to scroll together with the site list.
5. **Expect** Closing and reopening Jetpack app to **not** show the "Welcome to Jetpack" screen
6. **Expect** Logging out and closing Jetpack app app to **not** show the "Welcome to Jetpack" screen

### Preview

<details>
<summary> 🎬 Video preview of the scrolling behavior </summary>

https://user-images.githubusercontent.com/4588074/199570238-e4d32f67-7a3e-4fda-a604-be25c22f38d1.mp4
</details>

| Light | Dark |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/198845004-e930d72e-693b-4041-8ef6-86df5bfbdf28.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/198845002-dac704a2-2efe-4689-bdfe-b1cb6a32b935.png"> |

### Merge Strategy
1. Merge dependency PR into `trunk`: #17388
   ⚠️ Note: That PR also has dependencies.
3. Remove `Not Ready for Merge` flag
4. Merge this PR

## Regression Notes
1. Potential unintended areas of impact
   N/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

3. What automated tests I added (or what prevented me from doing so)
   None for now - planned for other PRs.

PR submission checklist:
